client/components/bulk-checkbox/index.js

```diff
@@ -11,9 +12,9 @@ const BulkCheckbox = ( { selectedCount, allCount, onToggle } ) => {
 	};
 
 	return (
-		<span className="wcc-bulk-checkbox-container" onClick={ onClick }>
-			<input type="checkbox" checked={ allSelected } readOnly />
-			{ someSelected ? <Gridicon className="bulk-select__some-checked-icon" icon="minus-small" size={ 18 }/> : null }
+		<span className="form-checkbox bulk-checkbox" onClick={ onClick }>
+			<Checkbox checked={ allSelected } readOnly />
+			{ someSelected && <Gridicon icon="minus-small" size={ 16 } /> }
```

Comment:
Can we move the `someSelected` logic to `Checkbox` directly, so only one component takes care of the entire visual state (unselected / partially selected / fully selected)?

The way it's written now the logic is a bit hard to follow, and also the HTML nesting is different in the two cases, making it a bit harder to change the CSS later.